CMakeLists.txt

```diff
@@ -21,7 +21,11 @@ set(CMAKE_CXX_STANDARD 20)
 set(CMAKE_CXX_EXTENSIONS ON)
 set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
 
-set(CMAKE_CXX_FLAGS "-fdiagnostics-color=always")
+if(APPLE) # TODO check to see if we are clang or g++
```

Comment:
I'm not sure how to use that syntax, can you propose the changes you would apply?

Reply:
It's a [generator expression](https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html#manual:cmake-generator-expressions(7)). It can be used directly in `target_compile_options` just like the existing `-Wno-register`